### ISSUE TITLE
M503: Print settings not implemented without eeprom compiled in on Due?

### ISSUE THREAD

**Issue by kyrreaa** [repetier/Repetier-Firmware#262]
Being able to read out the settings without eeprom compiled in is important.
It allows you to verify the correct image has been loaded...
It's easy to load the wrong hex, and reading the options will not be possible without eeprom compiled in on the Arduino Due.  This possibly also applies to the AVR-based implementation but I did not check.

**Comment by Engineer3DPrototyping** [comment]
Repetier by any chance did the eeprom simulator ever actually get implemented for the due? I ask because I an ultratronics pro board and it has everything I need except eeprom which is a really nice feature that I have been missing sense I upgraded from a RUMBA board. 
Alternatively is there any way to merge the SDEEPROM feature from the Davinci firmware replacement from luc that emulates eeprom with a sd card because this particular board has an intergrated sd card reader. Thank you for your time and effort.

**Comment by repetier** [comment]
I'm not really a friend of using flash as replacement especially as due erases it every upload and the number of writes is limit until it is destroyed and we save every change, so 1000 changes are  easy to achieve while eeproms last 100000 or 1000000 writes. sd card sounds like a nice alternative. It is cheap to replace and we have eeprom in ram for due, so it is just writing the 4096 byte on a command.

Can you show me this sdeeprom solution or say what codes it uses? Seems like you have some knowledge about this already. Backporting should be easy.

**Comment by Engineer3DPrototyping** [comment]
The feature that I would like to bring into the dev branch is called SDEEPROM from 
 https://github.com/luc-github/Repetier-Firmware-0.92/tree/0.92.6-sync
the files that appear to contain code for this particular feature are  
HAL.h
Printer.cpp
Repetier.h
SDCard.cpp
userpins.h
Commands.cpp
Eeprom.cpp
HAL.cpp
I can "kind" of follow allow and understand what Luc has done but his "fork's" Eeprom.cpp uses EEPROM version 12 and dev's is version 16 and I have not been able to figure out how to get the two to merge properly. 
Any help or advice would be of greatly appreciated. I have used luc's fork on a davinci 2.0 duo to upgrade it and free it from the chipped filament cartridges, and it worked perfectly. It wasn't until I tried to use the dev branch on the ultratronics that I discovered what a neat feature luc has developed.  Again Thank you for your time and effort.

**Comment by Engineer3DPrototyping** [comment]
Yes I agree but I plan on leaving an sd card in and using either repetier host or server to save gcode files to it because the due has native usb speeds and the ultratronics has the hardware for Ethernet.
